Mathlib/RingTheory/LaurentSeries.lean

```diff
@@ -429,3 +432,55 @@ instance : IsScalarTower F[X] (RatFunc F) (LaurentSeries F) :=
 end RatFunc
 
 end RatFunc
+
+section AdicValuation
+
+namespace PowerSeries
+
+variable (K : Type*) [Field K]
+
+/-- The prime ideal `(X)` of `PowerSeries K`, when `K` is a field, as a term of the
+`HeightOneSpectrum`. -/
+def idealX : IsDedekindDomain.HeightOneSpectrum K⟦X⟧ where
+  asIdeal := Ideal.span {X}
+  isPrime := PowerSeries.span_X_isPrime
+  ne_bot  := by rw [ne_eq, Ideal.span_singleton_eq_bot]; exact X_ne_zero
+
+open RatFunc IsDedekindDomain.HeightOneSpectrum
+
+variable {K}
+
+/- The `X`-adic valuation of a polynomial equals the `X`-adic valuation of its coercion to `K⟦X⟧`-/
+theorem intValuation_eq_of_coe (P : K[X]) :
+    (Polynomial.idealX K).intValuation P = (idealX K).intValuation (↑P : K⟦X⟧) := by
```

Comment:
```suggestion
    (idealX K).intValuation P = (idealX K).intValuation (↑P : K⟦X⟧) := by
```

Reply:
No, unfortunately this does not work; Lean thinks the first is the ideal in the power series and automatically coerces `P` there.